src/librustc_typeck/check/mod.rs

```diff
@@ -3858,13 +3858,9 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                           element_ty
                       }
                       None => {
-                          let mut err = self.type_error_struct(
-                              expr.span,
-                              |actual| {
-                                  format!("cannot index a value of type `{}`",
-                                          actual)
-                              },
-                              base_t);
+                          let mut err = struct_span_err!(tcx.sess, expr.span, E0608,
+                                                         "cannot index a value of type `{}`",
```

Comment:
That's not the into the problem, that's the index. I think switching the message to this would make the error lose its meaning.